subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/ivyservice/resolveengine/graph/builder/ModuleResolveState.java

```diff
@@ -55,6 +57,8 @@
     private final IdGenerator<Long> idGenerator;
     private final ModuleIdentifier id;
     private final List<EdgeState> unattachedDependencies = new LinkedList<EdgeState>();
+    private final Set<ModuleResolveState> parents = Sets.newHashSet();
```

Comment:
:o: mmm, do we have to track all parents? It seems quite expensive. Also, wouldn't we need to remove from parents at some point? Edge states are added/removed, and _they_ reference the parents already, no?

Reply:
This is now removed